### PR TITLE
session: adjust the upgrade compatibility behavior

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/46639

Problem Summary:

### What is changed and how it works?
Whether or not the `/upgrade/start` request for the old TiDB is received determines whether user DDL needs to be paused during the upgrade.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/sync_upgrade.go [hunk 1]
```diff
@@ -135,19 +134,17 @@ func isUpgradingClusterStateWithRetry(s sessionctx.Context, oldVer, newVer int64
 	for i := 0; ; i++ {
 		isUpgrading, err := IsUpgradingClusterState(s)
 		if err == nil {
-			if isUpgrading {
-				break
-			}
-			logger.Fatal("global state isn't upgrading, please send a request to start the upgrade first", zap.Error(err))
+			logger.Info("get global state", zap.Int64("old version", oldVer), zap.Int64("latest version", newVer), zap.Bool("is upgrading state", isUpgrading))
```
Comment (Benjamin2037) [inline session/sync_upgrade.go:137]: if no err, will this log print too much?

Comment (zimulala) [reply session/sync_upgrade.go:137]: Just print the upgrade status and version information

Comment (Benjamin2037) [reply session/sync_upgrade.go:137]: I mean too frequently.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by easonn7** [comment]
/approve

There is no sysvar change.

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**